publishToConfluence task is not updating the pages, however the build isn't failing

**Describe the bug**
The publishToConfluence task is not updating our pages.
For pushing we are using images which are built on a weekly basis, so they are always pulling the latest version of doctoolchain during the build. We didn't realized this issue for a while, because the publish task is not failing, only the output changes and there are no updates to the pages (we are using doctoolchain to fetch data to documentation pages in confluence about the images built by us).
We tested this behaviour by using different older versions of our doctoolchain images, and if we use an image to publish with version v3.3.1 the push works but when we use another ones with newer versions it just not working.
We didn't change anything in our dockerfile, so there should be probably some changes in doctoolchain in the recent past, we also checked the changelogs but didn't find any clue. Can you suggest any fix for this, or point out what is failing? Probably we have to adjust our code to some changes.

**The output with --stacktrace for the working publish task:**
```
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details
> Task :core:compileJava NO-SOURCE
> Task :core:compileGroovy
> Task :core:processResources NO-SOURCE
> Task :core:classes
> Task :core:jar

> Task :generateHTML
Converting /tmp/build/4629b6c3/theDCWorkingDir/src/docs/service-api.adoc

> Task :publishToConfluence
publish build/docs/html5/service-api.html
mavenpersonalnexus-jdk8
.
pages retrieved
retrieving page with id 3277739673
found existing page: 3277739673 version 37
> updated page 3277739673
published to https://devstack.vwgroup.com/confluence//spaces/DEVICEPLAT/pages/1561847896

BUILD SUCCESSFUL in 4m 14s
4 actionable tasks: 4 executed
```

**The output with --stacktrace for the non working publish task (v3.4.0):**
```
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

> Configure project :
>>>>>>>>> fix grolifant
>>>>>>>>> fix okhttp-digest

> Task :core:compileJava NO-SOURCE
> Task :core:compileGroovy
> Task :core:processResources NO-SOURCE
> Task :core:classes
> Task :core:jar
> Task :generateHTML
Converting /tmp/build/4629b6c3/theDCWorkingDir/src/docs/service-api.adoc

> Task :publishToConfluence
publish build/docs/html5/service-api.html
Publish /tmp/build/4629b6c3/theDCWorkingDir/build/docs/html5/service-api.html to DEVICEPLAT at https://devstack.vwgroup.com/confluence/rest/api/ ...
published to https://devstack.vwgroup.com/confluence//spaces/DEVICEPLAT/pages/1561847896

BUILD SUCCESSFUL in 2m 22s
4 actionable tasks: 4 executed 
``` 

**Expected behavior**
The pages should be retrieved, then updated if there are any changes (as described in the attached output).

**Configuration**
 - docToolchain v3.3.1
 - OS: alpine linux
 - openjdk17

It seems that the page is not retrieved
If you need, we can also paste you the output with the --debug flag

Thank you in advance

I am trying to add docToolchain inside a regularly-rebuilt container used in a workflow to run publishToConfluence. My previous working container had a local install of v2.2.1, but now trying to install or build the Docker image for any version <=3.3.1 (including 2.2.1) causes the dependency issues listed [here](https://github.com/docToolchain/docToolchain/issues/1414), or below in the case of v2.2.1. However, upgrading to v3.4.0+ means publishToConfluence fails silently as mentioned in this thread. 

Have others encountered this? Is there a workaround I may be missing?

```
69.49 FAILURE: Build failed with an exception.
69.49
69.49 * Where:
69.49 Build file '/home/jenkins/docToolchain/build.gradle' line: 89
69.49
69.49 * What went wrong:
69.49 A problem occurred evaluating root project 'docToolchain'.
69.49 > Could not resolve all artifacts for configuration 'classpath'.
69.49    > Could not find org.codehaus.groovy.modules.http-builder:http-builder:0.7.2.
69.49      Searched in the following locations:
69.49        - https://plugins.gradle.org/m2/org/codehaus/groovy/modules/http-builder/http-builder/0.7.2/http-builder-0.7.2.pom
69.49      If the artifact you are trying to retrieve can be found in the repository but without metadata in 'Maven POM' format, you need to adjust the 'metadataSources { ... }' of the repository declaration.
69.50      Required by:
69.50          unspecified:unspecified:unspecified
```

--- a/core/src/main/groovy/org/docToolchain/tasks/Asciidoc2ConfluenceTask.groovy
+++ b/core/src/main/groovy/org/docToolchain/tasks/Asciidoc2ConfluenceTask.groovy
@@ -694,18 +694,18 @@ class Asciidoc2ConfluenceTask extends DocToolchainTask {
         def anchors = [:]
         def pageAnchors = [:]
         def pages = []
-        def sections = []
         def title = dom.select('h1').text()
         if (maxLevel <= 0) {
             dom.select('div#content').each { pageBody ->
                 pageBody.select('div.sect2').unwrap()
                 promoteHeaders pageBody, 2, 1
-                def page = [title   : title,
-                            body    : pageBody,
-                            children: [],
-                            parent  : parentId]
+                def page = [
+                    title: title,
+                    body: pageBody,
+                    children: [],
+                    parent: parentId
+                ]
                 pages << page
-                sections = page.children
                 parentId = null
                 anchors.putAll(parseAnchors(page))
             }
@@ -720,15 +720,20 @@ class Asciidoc2ConfluenceTask extends DocToolchainTask {
                     parent: parentId
                 ]
                 pages << preamble
-                sections = preamble.children
                 parentId = null
                 anchors.putAll(parseAnchors(preamble))
+                
+                // Direkte Manipulation der children-Liste
+                preamble.children.addAll(getPagesRecursive(dom, parentId, anchors, pageAnchors, 1, maxLevel))
             }
-            sections.addAll(getPagesRecursive(dom, parentId, anchors, pageAnchors, 1, maxLevel))
+            
+            // Falls keine Präambel gefunden wurde, füge Seiten direkt zu pages hinzu
+            if (pages.isEmpty()) {
+                pages.addAll(getPagesRecursive(dom, parentId, anchors, pageAnchors, 1, maxLevel))
+            }
         }
         return [pages, anchors, pageAnchors]
     }
-
     void execute() {
         if(config.confluence.inputHtmlFolder) {
             def htmlFolder = "${docDir}/${config.confluence.inputHtmlFolder}"

